Userland/Libraries/LibJS/Runtime/DataView.h

```diff
@@ -17,22 +18,32 @@ class DataView : public Object {
     JS_DECLARE_ALLOCATOR(DataView);
 
 public:
-    static NonnullGCPtr<DataView> create(Realm&, ArrayBuffer*, size_t byte_length, size_t byte_offset);
+    static NonnullGCPtr<DataView> create(Realm&, ArrayBuffer*, ByteLength byte_length, size_t byte_offset);
 
     virtual ~DataView() override = default;
 
     ArrayBuffer* viewed_array_buffer() const { return m_viewed_array_buffer; }
-    size_t byte_length() const { return m_byte_length; }
+    ByteLength const& byte_length() const { return m_byte_length; }
     size_t byte_offset() const { return m_byte_offset; }
 
 private:
-    DataView(ArrayBuffer*, size_t byte_length, size_t byte_offset, Object& prototype);
+    DataView(ArrayBuffer*, ByteLength byte_length, size_t byte_offset, Object& prototype);
 
     virtual void visit_edges(Visitor& visitor) override;
 
     GCPtr<ArrayBuffer> m_viewed_array_buffer;
-    size_t m_byte_length { 0 };
+    ByteLength m_byte_length { 0 };
```

Comment:
Sorry - one question here that has been confusing me just now - why is it we use `size_t` in some places when passing around byte length and `u32` in others? (not that I imagine such large allocations are made anyway)

I assume we're migrating to `u32` in LibJS? But why would `DataView` have a different type that `ArrayBuffer`? Just more migration work (which then led me down the path of `MAX_ARRAY_LIKE_INDEX` in `create_byte_data_block`)

Reply:
tbh I don't know. I picked `u32` for `ByteLength` because that is what `m_array_length` and `m_byte_length` were stored as in TypedArray.h, and (even though it's the last commit) that's where I started when writing this. Otherwise, in these patches I just matched the type that was nearest. For example, in `allocate_array_buffer`, there was an existing `size_t byte_length` parameter, so I made `max_byte_length` be `size_t` as well.

I know at one point we were trying to avoid `size_t` to avoid 32/64-bit differences. According to https://stackoverflow.com/a/72124984, `u32` would be okay to match both chrome and safari. We should probably do a follow-up to make these array classes all `u32`.

Reply:
Makes sense, thanks!